python/tests/test_deserialization.py

```diff
@@ -0,0 +1,1060 @@
+@pytest.mark.asyncio
+@pytest.mark.requires_db
+async def test_date_deserialization(session: Session, table_factory: TableFactory):
+    table = await table_factory("id int PRIMARY KEY, d date", "date_table")
+
+    # Prepare two dates: one recent, one historic
+    d1 = datetime.date(2024, 5, 10)
+    d2 = datetime.date(1999, 12, 31)
+
+    # Insert using ISO strings (Cassandra accepts YYYY-MM-DD)
+    await session.execute(f"INSERT INTO {table} (id, d) VALUES (1, '{d1.isoformat()}');")
+    await session.execute(f"INSERT INTO {table} (id, d) VALUES (2, '{d2.isoformat()}');")
+
+    # Read back
+    result = await session.execute(f"SELECT * FROM {table}")
+    rows = list(result.iter_rows())
+
+    assert len(rows) == 2
+
+    row1 = next(r for r in rows if r["id"] == 1)
+    row2 = next(r for r in rows if r["id"] == 2)
+
+    # Type checks
+    assert isinstance(row1["d"], datetime.date)
+    assert isinstance(row2["d"], datetime.date)
+
+    # Value correctness
+    assert row1["d"] == d1
+    assert row2["d"] == d2
+
+    # Additional sanity checks (same ordering)
+    assert row1["d"].isoformat() == "2024-05-10"
+    assert row2["d"].isoformat() == "1999-12-31"
+
+
+# Verifies correct deserialization of CQL ascii into Python strings
+@pytest.mark.asyncio
+@pytest.mark.requires_db
+async def test_ascii_deserialization(session: Session, table_factory: TableFactory):
+    table = await table_factory("id int PRIMARY KEY, txt ascii", "ascii_table")
+
+    # ASCII-only values
+    v1 = "HelloASCII"
+    v2 = "Test123_ABC"
+
+    # Insert values
+    await session.execute(f"INSERT INTO {table} (id, txt) VALUES (1, '{v1}');")
+    await session.execute(f"INSERT INTO {table} (id, txt) VALUES (2, '{v2}');")
+
+    # Read back
+    result = await session.execute(f"SELECT * FROM {table}")
+    rows = list(result.iter_rows())
+
+    assert len(rows) == 2
+
+    row1 = next(r for r in rows if r["id"] == 1)
+    row2 = next(r for r in rows if r["id"] == 2)
+
+    # Type checks
+    assert isinstance(row1["txt"], str)
+    assert isinstance(row2["txt"], str)
+
+    # Value correctness
+    assert row1["txt"] == v1
+    assert row2["txt"] == v2
+
+    # Ensures no accidental unicode alterations
+    assert row1["txt"].isascii()
+    assert row2["txt"].isascii()
+
+
+# Verifies exact round-trip deserialization of CQL decimal into Python Decimal
+@pytest.mark.asyncio
+@pytest.mark.requires_db
+@pytest.mark.parametrize(
+    "row_id,value",
+    [
+        (1, Decimal("0")),
+        (2, Decimal("0.0")),
+        (3, Decimal("-0")),
+        (4, Decimal("123.456")),
+        (5, Decimal("-7890.00123")),
+        (6, Decimal("123456789012345678901234567890.123456789")),
+        (7, Decimal("-999999999999999999999999999999999999.9999")),
+        (8, Decimal("0.00000000000000000001")),
+        (9, Decimal("42.000000")),
+        (10, Decimal("617283694")),
+        (11, Decimal("617283694.4324")),
+    ],
+)
+async def test_decimal_deserialization(
```

Comment:
You wrote this test using parametrize, but in all other tests you duplicated all test lines for all cases.

Please remove the duplication in other tests. We don't want to have to duplicate whole test of e.g. int just to add a test case.
Wheter you use `parametrize` or just an array defined inside test is up to you, I don't have enough knowledge about Python testing conventions to decide that.